server/src/test/java/org/elasticsearch/search/suggest/AbstractSuggestionBuilderTestCase.java

```diff
@@ -165,10 +165,7 @@ public void testBuild() throws IOException {
                     indexSettings);
             MapperService mapperService = mock(MapperService.class);
             ScriptService scriptService = mock(ScriptService.class);
-            boolean fieldTypeSearchAnalyzerSet = randomBoolean();
```

Comment:
wasn't this a useful test? Dont we lose coverage by removing this case? maybe though we should have a specific test for this scenario?

Reply:
The test is checking what happens when a field has a `null` analyzer configured.  The only place that this actually happens is in this test (I have a separate PR ready to go that adds null checks to TextSearchInfo, and the only place that triggers failures is this test), so it's in effect testing a situation that will never occur.

Reply:
thanks for explaining, I thought that it was rather testing the fallback from search analyzer to default search analyzer.

Reply:
Maybe that is what the test wanted to initially do but it ended up setting a null analyzer to TextSearchInfo which would not be allowed?

Reply:
yes, the 'fallback-to-default' behaviour is now done up-front in TextSearchInfo, so it can't be tested here anymore.

Reply:
yes yes I finally got it. Last famous words :)